### DISCUSSION TITLE
SSR: Why are page props output as json inside `<script type="application/json" sveltekit:data-type="props">`instead of the hydrate script?

### DISCUSSION THREAD

**Discussion by manvankaliber** [sveltejs/kit#4899] [Q&A]
I do not understand the reason that svelte-kit includes page props separate script tag as a json string, instead of it being included as a javascript object in the hydrate script tag or as a global constant available tot the page. Would that not be more compress-able and easier to access in the case you need it?

**Comment by mrkishi** [comment]
I don't know the specific reasons for it, but I wouldn't be surprised if it was due to prior art or previous attempts at serializing more arbitrary objects (eg. in Sapper).

While it'd be possible to move it to the script tag, we'd actually gain very little:

- It's unlikely it'd be more compressible in any way since we'd still need to keep them as JSON strings for security reasons (there's no way to safely render a javascript object to be used as-is). In fact, it'd probably be _less_ compressible since we'd have to escape at least one extra character.
- While it'd be slightly "easier" to access, we'd be responsible for making sure they were properly disposed of, since if we added them to the script naively like `const ssr_data = ...` they'd live in the JS heap past their use (unlike today). I'd call it even on this point.

Having said all that, I wouldn't be against it, but maybe there were more important reasons for the tag approach I'm not aware of.

**Comment by Rich-Harris** [comment]
It's primarily for performance reasons: https://www.bram.us/2019/11/25/faster-javascript-apps-with-json-parse/

**Reply by mrkishi** [reply]
In that case we _could_ move it to the script, since we would still need to have it as a JSON string anyway for security reasons. But I don't think it'd be particularly better, tbh.
